src/hagl_hal/LICENSE

```diff
@@ -0,0 +1,21 @@
+MIT License
```

Comment:
I also need to add a license. My code will also be MIT

Reply:
also I should probably move all these libraries to external/ with tinyusb. I modified all of them in order to get them to work so I can't use git-submodule, but they are effectively third-party libraries anyways